### ISSUE TITLE
Inverse Spectrogram and Mel-Spectrogram Layer?

### ISSUE THREAD

**Issue by AI-Guru** [keunwoochoi/kapre#40]
Namaste!

kapre has become an integral part of all my audio Deep Learning experiments. Powerful! Thanks for providing such a great software!

I was thinking... I guess it would make sense to have layers for inverse spectrogram and inverse mel-spectrogram. Thinking about Autoencoders, this would be even more powerful. I know that reconstructing samples from spectrograms is not the best, but it is possible to a certain degree.

What do you think about that feature request?

Best,
Tristan

**Comment by keunwoochoi** [comment]
Hi, good to know that Kapre is useful for you! I think that's good idea, although the development of Kapre has been paused for a while. I'll get back to work in a month and will make some update then.

**Comment by douglas125** [comment]
Maybe make a Griffin-Lim spectrogram inversion?

**Comment by jonnor** [comment]
Btw, librosa 0.7+ has Griffin-Lim and inverse melspectrogram. Not using GPU though. https://librosa.github.io/librosa/generated/librosa.feature.inverse.mel_to_audio.html#librosa.feature.inverse.mel_to_audio

**Comment by AI-Guru** [comment]
Wow! Is there a way to push a gradient through this?

**Comment by jonnor** [comment]
Forward pass probably not a problem. No idea about backpropagation though. Probably hard, as Griffin-Lim is a iterative algorithm. One-shot approximations for spectrogram inversion based on neural networks have been explored in the literature though, for example [Fast Spectrogram Inversion using Multi-head Convolutional Neural Networks](https://arxiv.org/abs/1808.06719). Those would be backprop-able by construction

**Comment by HudsonHuang** [comment]
Maybe we have STFT before [this line](https://github.com/keunwoochoi/kapre/blob/8517f45d3ccb9fff1ec0049b9c3e4389f81c20aa/kapre/time_frequency.py#L175) ? 
So, the inverse of Spectrogram and Mel-Spectrogram might be diffcult, but the inverse of STFT might be easy? A pytorch version is [here](https://github.com/diggerdu/pytorch_audio/blob/master/time_frequence.py#L68)

**Comment by keunwoochoi** [comment]
It's on my to-do now and will be added on `0.3.x`.
